File: src/main/kotlin/com/tezal/hadith/controller/AuthController.java
package com.tezal.hadith.controller;

import com.tezal.hadith.model.dto.UserDto;
import com.tezal.hadith.service.AuthService;
import io.swagger.annotations.ApiParam;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/auth")
public class AuthController {
    private final AuthService authService;

    public AuthController(AuthService authService) {
        this.authService = authService;
    }
    @PostMapping("/signin")
    public String login(@ApiParam("Username") @RequestParam String username,
                        @ApiParam("Password") @RequestParam String password) {
        return authService.signin(username, password);
    }
    @PostMapping("/signup")
    public String signUP(@RequestBody UserDto userDto) {
        return authService.signUp(userDto);
    }



}

File: src/main/kotlin/com/tezal/hadith/controller/AuthController.kt
package com.tezal.hadith.controller

import com.tezal.hadith.model.auth.AuthRequest
import com.tezal.hadith.model.auth.AuthResponse
import com.tezal.hadith.service.AuthService
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/auth")
class AuthController(private val authService: AuthService) {
    @PostMapping("/signin")
    fun login(@RequestBody user: AuthRequest): AuthResponse {
        return AuthResponse(authService.signin(user.username, user.password))
    }

//    @PostMapping("/signup")
//    fun signUP(@RequestBody userDto: UserDto?): String {
//        return authService.signUp(userDto)
//    }
}